packages/driver/src/cy/errors.coffee

```diff
@@ -1,16 +1,24 @@
 $dom = require("../dom")
 $utils = require("../cypress/utils")
+_ = require('lodash')
 
 crossOriginScriptRe = /^script error/i
 
 create = (state, config, log) ->
   commandErr = (err) ->
     current = state("current")
 
+    args = current.get("args")
+    options = null
+
+    if _.isObject(args[args.length - 1])
+      options = args[args.length - 1]
+
```

Comment:
this assumes the last argument is `options` if it's an object, I'm not sure that's always true

Reply:
True. Although this assumption is true for [the API of 3.8.3](https://docs.cypress.io/api/api/table-of-contents.html), it won't be true in the future. Or it might not be true for custom commands. 

But I cannot think of a good alternative.

Reply:
I found the cases that don't meet the assumption: 

```js
cy.clearLocalStorage(/app-/)
cy.clock(null, ['setTimeout', 'clearTimeout'])
```

`/app-/` and `['setTimeout', 'clearTimeout']` are the last argument and an object. But it is not an option. 

I think we need a function like `getUserOptions(name, args)` that handles these special cases. and change the code like below: 

```js
options = getUserOptions(current.get('name'), current.get("args"))
```

 Or is there a better way?